core/utilities-ast.lua

```diff
@@ -0,0 +1,213 @@
+--- SILE AST utilities
+--
+local ast = {}
+
+--- Find a command node in a SILE AST tree,
+--- looking only at the first level.
+--- (We're not reimplementing XPath here.)
+---@param tree      table       AST tree
+---@param command   string      command name
+---@return          table|nil   AST command node
+function ast.findInTree (tree, command)
+  for i=1, #tree do
+    if type(tree[i]) == "table" and tree[i].command == command then
+      return tree[i]
+    end
+  end
+end
+
+--- Find and extract (remove) a command node in a SILE AST tree,
+--- looking only at the first level.
+---@param tree      table       AST tree
+---@param command   string      command name
+---@return          table|nil   AST command node
+function ast.extractFromTree (tree, command)
```

Comment:
I don't think "extract" is the right verb here, at least to me that implies getting a node out in isolation, but also I would have assumed it *left* the original tree alone. This removes it from the tree, so "slice" or perhaps "extricate" might be a better name that suggests the original is being removed.

Reply:
maybe simpy `removeFromTree` (and the `@return` becomes "Removed AST command node") ?